# Strukturální fondy
Loni jsem [se hrabal](https://gist.github.com/kokes/052cc2a63e6366fc92b1) v datech od MMR, konkrétně v tabulce projektů za období 2007-2013. A jelikož to [nikoho nezajímalo](https://twitter.com/kondrej/status/604406720471961600), rozhodl jsem se to udělat znovu a pořádněji.

Vše níže uvedné je na základě [dat na webu MMR](http://www.strukturalni-fondy.cz/cs/Informace-o-cerpani/Seznamy-prijemcu). Rozdíl od loňských hrátek je v tom, že MMR v mezičase dodalo mnohem čistší tabulku. Dříve člověk musel párovat projekty na vládní instituce, NACE kategorie atd. Teď to za nás udělalo ministerstvo, takže můžeme stratifikaci prohlížet mnohem snadněji.

Stížnosti do [twitterů](http://twitter.com/kondrej) nebo [mailů](mailto:ondrej.kokes@gmail.com)

*(Kód používá víceméně jen Python 3 a knihovnu `pandas`, klidně si to pusťte)*

In [1]:
import pandas as pd
import numpy as np
import urllib.request
import os
%matplotlib inline

In [62]:
url = 'http://www.strukturalni-fondy.cz/getmedia/d10fe3b0-15f5-4cdb-b916-64be5b4c5c4d/Prehled_projektu_20160406u.xlsx?ext=.xlsx'
fn = 'vstupy/prehled.xlsx'
dvs = 'vystupy' # kam se budou ukladat vystupy

if not os.path.exists(fn):
    urllib.request.urlretrieve(url, fn)

In [3]:
df = pd.read_excel(fn)
df = df.loc[df.loc[:, 'Pořadí v rámci v projektu (filtr)'] == 1] # kazdy projekt ma vic radku

Celkem máme 75 tisíc projektů z období 2007-2013, ke každému 28 indikátorů, většina je celkem sebevysvětlující.

In [4]:
df.shape

(75137, 28)

In [5]:
df.columns

Index(['Číslo a název programu', 'Číslo prioritní osy',
       'Číslo oblasti podpory', 'Číslo projektu', 'Název projektu',
       'Popis projektu', 'Žadatel', 'IČ žadatele', 'Hosp.právní forma',
       'Hosp.právní forma - skupina', 'Stav projektu',
       'Datum podpisu smlouvy/rozhodnutí', 'Adresa žadatele',
       'Kraj žadatele - kód', 'Kraj žadatele - název', 'Obec žadatele - kód',
       'Obec žadatele - název', 'Celkové zdroje - Sml/Dod',
       'Rozhodnutí/Smlouva o poskytnutí dotace (celkové zdroje)',
       'Rozhodnutí/Smlouva o poskytnutí dotace (veřejné prostředky celkem)',
       'Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)',
       'Proplacené prostředky příjemcům - vyúčtované (veřejné prostředky celkem)',
       'Proplacené prostředky příjemcům - vyúčtované (EU zdroje)',
       'Certifikované prostředky (veřejné prostředky celkem)',
       'Certifikované prostředky (EU zdroje)', 'Název NUTS',
       'Pořadí v rámci v projektu (filtr)', 'Počet míst realizace'],
  

In [6]:
# nazvy sloupcu, at se to lip filtruje
prfo = 'Hosp.právní forma'
prfos = 'Hosp.právní forma - skupina'
prog = 'Číslo a název programu'
nuts = 'Název NUTS'
ico = 'IČ žadatele'
zad = 'Žadatel'
stav = 'Stav projektu'
popis = 'Popis projektu'
nazev = 'Název projektu'
celk = 'Celkové zdroje - Sml/Dod'
euz = 'Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)'

***V datasetu od MMR je několik metrik pro částku hrazenou z dotací. Jde o Rozhodnutí, Proplacené prostředky a Certifikované prostředky. Je tak možné měřit objem dotací více způsoby. V celém tomto dokumentu se měří pouze ono Rozhodnutí. Změnou proměnné `euz` výše tuto funkcionalitu změníte.***

První tři projekty vypadají takto

In [7]:
df.head(3).T

0   \
Číslo a název programu                                                             CZ.1.01 OP Doprava   
Číslo prioritní osy                                                                               1.1   
Číslo oblasti podpory                                                                           1.1.1   
Číslo projektu                                                                 CZ.1.01/1.1.00/06.0019   
Název projektu                                         Optimalizace trati Benešov u Prahy - Strančice   
Popis projektu                                      Délka rekonstruovaného úseku je 23,95 km. Stáv...   
Žadatel                                             Správa železniční dopravní cesty, státní organ...   
IČ žadatele                                                                                  70994234   
Hosp.právní forma                                                               Správa žel.dopr.cesty   
Hosp.právní forma - skupina                                        Stát a jeho instituce a organizace   
Stav projektu                                                        P7 Výdaje projektu certifikovány   
Datum podpisu smlouvy/rozhodnutí                                                  2008-07-02 00:00:00   
Adresa žadatele                                                          11000 Praha, Dlážděná 1003/7   
Kraj žadatele - kód                                                                             CZ010   
Kraj žadatele - název                                                              Hlavní město Praha   
Obec žadatele - kód                                                                      CZ0100554782   
Obec žadatele - název                                                                           Praha   
Celkové zdroje - Sml/Dod                                                                  4.04112e+09   
Rozhodnutí/Smlouva o poskytnutí dotace (celkové...                                        3.70623e+09   
Rozhodnutí/Smlouva o poskytnutí dotace (veřejné...                                        2.60026e+09   
Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)                                        3.65886e+09   
Proplacené prostředky příjemcům - vyúčtované (v...                                        2.56681e+09   
Proplacené prostředky příjemcům - vyúčtované (E...                                        3.65886e+09   
Certifikované prostředky (veřejné prostředky ce...                                        2.56681e+09   
Certifikované prostředky (EU zdroje)                                                     CZ0201529303   
Název NUTS                                                                                    Benešov   
Pořadí v rámci v projektu (filtr)                                                                   1   
Počet míst realizace                                                                               10   

                                                                                                   10  \
Číslo a název programu                                                             CZ.1.01 OP Doprava   
Číslo prioritní osy                                                                               1.1   
Číslo oblasti podpory                                                                           1.1.1   
Číslo projektu                                                                 CZ.1.01/1.1.00/07.0006   
Název projektu                                      Optimalizace trati Horní Dvořiště st. hranice ...   
Popis projektu                                      Vzhledem k již proběhlé částečné optimalizaci ...   
Žadatel                                             Správa železniční dopravní cesty, státní organ...   
IČ žadatele                                                                                  70994234   
Hosp.právní forma                                                               Správa žel.dopr.cesty   
Hos

### Další datasety na napojení - veřejné podniky
Sice náš MMR dataset obsahuje informace o státních institucích, seznam není kompletní. Podniky s výrazným podílem státu nebo firmy plně vlastněné kraji, obcemi atd., jsou v datasetu často označené jako běžné akciové společnosti.

Můžeme ale použít dva datasety of ministerstva financí:
- [Seznam vládních institucí v ČR](http://www.mfcr.cz/cs/verejny-sektor/rozpoctove-ramce-statisticke-informace/verejny-sektor/sektor-vladnich-instituci/2016/seznam-vladnich-instituci-v-cr-2016-24749)
- [Seznam veřejných společností v ČR](http://www.mfcr.cz/cs/verejny-sektor/rozpoctove-ramce-statisticke-informace/verejny-sektor/verejne-spolecnosti/2016/seznam-verejnych-spolecnosti-v-cr-2016-24752)

In [8]:
url = 'http://www.mfcr.cz/assets/cs/media/Rozp-ramce-EU-85-2011_2016_Seznam-verejnych-spolecnosti-v-CR.xlsx'
fn = 'vstupy/verejne-instituce.xlsx'

if not os.path.exists(fn):
    urllib.request.urlretrieve(url, fn)
    
url = 'http://www.mfcr.cz/assets/cs/media/Rozp-ramce-EU-85-2011_2016_Seznam-vladnich-instituci-v-CR.xls'
fn = 'vstupy/vladni-instituce.xls'

if not os.path.exists(fn):
    urllib.request.urlretrieve(url, fn)

In [9]:
ver = pd.read_excel('vstupy/verejne-instituce.xlsx', skiprows=4)
ver = ver.iloc[:, 1:]
ver.head(5)

IČO                              NÁZEV DATUM VZNIKU   FORMA   NACE  \
0  46967036               JUPITER club, s.r.o.    1992-08-03    112  90040   
1  48909467  Služby města Konice, spol. s r.o.    1993-08-18    112  38110   
2  25013599                   TS HRÁDEK s.r.o.    1996-12-19    112  38110   
3  25013866        Klášterecká kyselka, s.r.o.    1996-12-31    112  47190   
4  25020307        Podnik služeb Jirkov s.r.o.    1997-05-06    112  81290   

   SUBSEKTOR  
0      11001  
1      11001  
2      11001  
3      11001  
4      11001

In [10]:
vlad = pd.read_excel('vstupy/vladni-instituce.xls', skiprows=4)
vlad = vlad.iloc[:, 1:]
vlad.head(5)

IČO                         NÁZEV DATUM VZNIKU  FORMA   NACE  COFOG  \
0  26925974     Nemocnice Boskovice s.r.o   2004-04-22    112  86100    NaN   
1   3960765            Lesy Drmoul s.r.o.   2015-04-03    112   2100    NaN   
2  27577708         Divadla Kladno s.r.o.   2006-08-02    112  90020    NaN   
3  26076357  Strakonická televize, s.r.o.   2004-07-10    112  60200    NaN   
4   4510984   Pošumavská odpadová, s.r.o.   2015-10-26    112     38    NaN   

   SUBSEKTOR  
0      13130  
1      13130  
2      13130  
3      13130  
4      13130

In [11]:
len(ver), len(vlad)

(1172, 18117)

In [12]:
vvico = list(ver.loc[:, 'IČO'].values) + list(vlad.loc[:, 'IČO'].values)
vvico = [str(l).rjust(8, '0') for l in vvico]
vvico = set(vvico)

## První prohrab

Hlavní přídavek oproti předchozímu datasetu je přítomnost klasifikací dle pravních forem.

In [47]:
df.loc[:, prfo].unique()[:10]

array(['Správa žel.dopr.cesty',
       'Správa železniční dopravní cesty, státní organizace',
       'Společnost s r.o.', 'Akciová společnost', 'Příspěvková organizace',
       'Státní příspěvková organizace', 'Kraj', 'Dobrovolný svazek obcí',
       'Obec nebo m.část hl.m.', 'Org. složka státu'], dtype=object)

Ty jsou seskupené do skupin. Nejvíc projektů (ne nutně peněz) připadá na obce.

In [48]:
df.loc[:, prfos].value_counts()

Obce                                   25192
Podnikatelské subjekty                 22118
Ostatní                                15068
Komory, profesní a zájmové sdružení     3221
Stát a jeho instituce a organizace      2542
Neziskový sektor                        1601
Kraje                                   1584
Vzdělávací a výzkumné instituce         1550
Name: Hosp.právní forma - skupina, dtype: int64

## Očista

### Nesprávné IČO
Je spousta projektů, kde IČO není správné české IČO. U zhruba čtyř tisíc případů IČO chybí, u cca 1800 jde o zahraniční IČO (většina projektů ze spolupráce Česko-Polsko).

In [29]:
df.loc[:, ico].isnull().sum()

4131

In [24]:
nims = df.loc[:, ico].apply(lambda x: len(str(x)) == 8)
len(df)-nims.sum()

5799

Celkem jde o projekty za skoro 19 miliard

In [35]:
(df.loc[~nims, euz].sum()/10**9).round(2)

18.84

O tyto projekty náš dataset očistíme.

In [36]:
df = df.loc[nims,:]

### Duplikované skupiny

In [37]:
df.loc[:, prfo] = df.loc[:, prfo].replace('Společen.vlast.jednot.', 'Společenství vlastníků jednotek')
df.loc[:, prfo] = df.loc[:, prfo].replace('Reg.rada regionu soudr.', 'Regionální rada regionu soudržnosti')

In [38]:
df.loc[:, prfo] = df.loc[:, prfo].replace('Vysoká škola (veřejná, státní)', 'Vysoká škola')

In [39]:
# df.loc[df.loc[:, prfo].str.contains('Vysok'), prfo].unique()

In [40]:
df.loc[:, prfo] = df.loc[:, prfo].replace('Příspěvková organizace', 'Státní příspěvková organizace')

In [41]:
df.loc[:, prfo] = df.loc[:, prfo].replace('Veř. výzkumná instituce', 'Veřejná výzkumná instituce')

### Veřejné instituce
Čtyři skupiny institucí patří mezi veřejné - obce, kraje, stát, vzdělávací a výzkumné instituce.

In [42]:
erar = 'veřejná instituce'
df.loc[:, erar] = False
df.loc[df.loc[:, prfos].isin(['Obce', 'Kraje', 'Stát a jeho instituce a organizace', 'Vzdělávací a výzkumné instituce']), erar] = True

...ale v 'Ostatní' je i několik skupin, které jsou veřejné.

In [43]:
list(df.loc[df.loc[:, prfos] == 'Ostatní', prfo].unique())

['Státní příspěvková organizace',
 'Příspěvková organizace zřízená územním samosprávným celkem',
 'Ústav',
 'Odštěp.záv.zaps.v ob.r.',
 'Společenství vlastníků jednotek',
 'Regionální rada regionu soudržnosti',
 'Zatím neurčeno']

In [44]:
df.loc[df.loc[:, prfo].isin(['Státní příspěvková organizace',
 'Příspěvková organizace zřízená územním samosprávným celkem',
 'Regionální rada regionu soudržnosti']), erar] = True

Taky je pár veřejných institucí mezi žadateli, kteří nemají hlavní nadskupinu.

In [45]:
df.loc[df.loc[:, prfos].isnull(),prfo].unique()[:6]

array(['Veřejná výzkumná instituce', 'Školská právnická osoba',
       'Evropská společnost',
       'Odborová organizace a organizace zaměstnavatelů', 'Ostatní',
       'Zem.podnik.FO nez.v OR'], dtype=object)

In [46]:
df.loc[df.loc[:,prfo].isin(['Veřejná výzkumná instituce', 'Školská právnická osoba']), erar] = True

Poslední zdroj informací budou ty dva datasety of MFČR, které jsme načetli výše. Projdeme tedy podniky, které jsou označené jako soukromé a podíváme se, jestli jeji IČO není v těch dvou datasetech.

In [47]:
dd = df.loc[df.loc[:,erar] == False, [ico, erar]]
for j in dd.iterrows():
    if j[1].loc[ico] not in vvico: continue
    df.loc[j[0], erar] = True

### EIB
Poslední výjimka, v datech je pod IČO 00000000 Evropská investiční banka, EIB. Tak ji přepneme do veřejných.

In [48]:
df.loc[df.loc[:, ico] == '00000000']

Číslo a název programu Číslo prioritní osy  \
100818  CZ.1.10 ROP NUTS II Moravskoslezsko                10.2   

       Číslo oblasti podpory           Číslo projektu  \
100818                10.2.3  CZ.1.10/2.3.00/02.00976   

                                 Název projektu  \
100818  JESSICA Holdingový fond Moravskoslezsko   

                                           Popis projektu  \
100818  Evidence financování JESSICA Holdingového fond...   

                          Žadatel IČ žadatele Hosp.právní forma  \
100818  Evropská investiční banka    00000000  Zahraniční osoba   

       Hosp.právní forma - skupina        ...          \
100818      Podnikatelské subjekty        ...           

       Rozhodnutí/Smlouva o poskytnutí dotace (veřejné prostředky celkem)  \
100818                                        430882000.0                   

       Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)  \
100818                                        506920000.0   

       Proplacené prostředky příjemcům - vyúčtované (veřejné prostředky celkem)  \
100818                                        430882000.0                         

       Proplacené prostředky příjemcům - vyúčtované (EU zdroje)  \
100818                                        506920000.0         

       Certifikované prostředky (veřejné prostředky celkem)  \
100818                                        430882000.0     

       Certifikované prostředky (EU zdroje)     Název NUTS  \
100818                         CZ0802598003  Frýdek-Místek   

        Pořadí v rámci v projektu (filtr)  Počet míst realizace  \
100818                                1.0                   5.0   

        veřejná instituce  
100818              False  

[1 rows x 29 columns]

In [49]:
df.loc[df.loc[:, ico] == '00000000', erar] = True

### Kofinancování

In [50]:
kofi = 'poměr vlastních financí (0-1)'
df.loc[:, kofi] = np.nan
fr = 1-df.loc[:, euz] / df.loc[:, celk]

fr = fr[fr >= 0] # mám pocit že u těch českopolských projektů je celková cena v eurech, ale dotace v korunách (a zlotých)
df.loc[fr.index, kofi] = fr.values

### Čištění o.p.s.

ops bylo někde špatně

In [51]:
df.loc[:, prfo] = df.loc[:, prfo].replace('Obecně prospěšná spol.', 'Obecně prospěšná společnost')

### Negativní dotace

**Negativní dotace.** Nezkoumal sem, jestli chyba nebo pokuta nebo co.

In [52]:
df.loc[df.loc[:, euz]<0, [zad, euz]]

Žadatel  \
79295   Odborná střední škola podnikatelská Kolín s.r.o.   
112983                    AAA BYTY.CZ akciová společnost   

        Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)  
79295                                       -5.820766e-11   
112983                                      -2.849530e+05

### Duplikovaní žadatelé
Máme víc žadatelů než IČO.

In [53]:
print(len(df.loc[:, zad].unique()))


print(len(df.loc[:, ico].unique()))

26229
24338


In [55]:
dd = df.loc[:, [zad,ico]].drop_duplicates().sort_values(by=ico)
dd.index = range(len(dd))
dd.head(3)

Žadatel IČ žadatele
0          Evropská investiční banka    00000000
1    Vojenské lesy a statky ČR, s.p.    00000205
2  Česká agentura na podporu obchodu    00001171

In [57]:
df.loc[:, [zad, ico]].drop_duplicates().shape

(26866, 2)

In [58]:
dd = df.loc[:, [zad, ico]].drop_duplicates()
dd = dd.loc[dd.duplicated(subset=ico, keep=False)].sort_values(by=ico)
dd.head(10)

Žadatel IČ žadatele
54040                 MINISTERSTVO FINANCÍ    00006947
54269                 Ministerstvo financí    00006947
6075   ČESKÁ SPRÁVA SOCIÁLNÍHO ZABEZPEČENÍ    00006963
10340  Česká správa sociálního zabezpečení    00006963
50573                  MINISTERSTVO VNITRA    00007064
7752                   Ministerstvo vnitra    00007064
54096         STÁTNÍ VETERINÁRNÍ SPRÁVA ČR    00018562
54236            Státní veterinární správa    00018562
54001             MINISTERSTVO ZEMĚDĚLSTVÍ    00020478
54267             Ministerstvo zemědělství    00020478

Můžem nahradit duplikovaný názvy.

In [59]:
for j in dd.drop_duplicates(subset=ico).iterrows():
    df.loc[df.loc[:, ico] == j[1].loc[ico], zad] = j[1].loc[zad]

**Furt nějaký zbývaj**

In [60]:
print(len(df.loc[:, zad].unique()))

print(len(df.loc[:, ico].unique()))

23860
24338


Bude možná tím, že v datech jsou mezery.

## Operační programy

In [64]:
prdf = df.loc[:, [prog, euz, erar]].groupby([prog, erar]).sum()
prdf /= 10**6
prdf = prdf.round(1).sort_values(by=euz, ascending=False).unstack()
prdf

Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)  \
veřejná instituce                                                                          False   
Číslo a název programu                                                                             
CZ.1.01 OP Doprava                                                                         198.1   
CZ.1.02 OP Životní prostředí                                                             20809.5   
CZ.1.03 OP Podnikání a inovace                                                           83765.9   
CZ.1.04 OP Lidské zdroje a zaměstnanost                                                  15346.1   
CZ.1.05 OP Výzkum a vývoj pro inovace                                                     4631.9   
CZ.1.06 Integrovaný operační program                                                      5483.1   
CZ.1.07 OP Vzdělávání pro konkurenceschopnost                                             9282.3   
CZ.1.08 OP Technická pomoc                                                                 106.5   
CZ.1.09 ROP NUTS II Severozápad                                                           1873.1   
CZ.1.10 ROP NUTS II Moravskoslezsko                                                       1766.8   
CZ.1.11 ROP NUTS II Jihovýchod                                                            2061.1   
CZ.1.12 ROP NUTS II Střední Morava                                                        2586.0   
CZ.1.13 ROP NUTS II Severovýchod                                                          2804.4   
CZ.1.14 ROP Jihozápad                                                                     1948.4   
CZ.1.15 ROP NUTS II Střední Čechy                                                         1252.6   
CZ.1.25 OP Rybářství                                                                       543.9   
CZ.2.16 OP Praha Konkurenceschopnost                                                       929.5   
CZ.2.17 OP Praha Adaptibilita                                                             1800.9   
CZ.3.22 OP ČR - Polsko                                                                      93.1   

                                                         
veřejná instituce                                 True   
Číslo a název programu                                   
CZ.1.01 OP Doprava                             191090.5  
CZ.1.02 OP Životní prostředí                   119224.1  
CZ.1.03 OP Podnikání a inovace                   3403.3  
CZ.1.04 OP Lidské zdroje a zaměstnanost         40853.6  
CZ.1.05 OP Výzkum a vývoj pro inovace           50119.7  
CZ.1.06 Integrovaný operační program            39111.9  
CZ.1.07 OP Vzdělávání pro konkurenceschopnost   41291.8  
CZ.1.08 OP Technická pomoc                       3980.3  
CZ.1.09 ROP NUTS II Severozápad                 19759.8  
CZ.1.10 ROP NUTS II Moravskoslezsko             20866.1  
CZ.1.11 ROP NUTS II Jihovýchod                  20891.7  
CZ.1.12 ROP NUTS II Střední Morava              18645.5  
CZ.1.13 ROP NUTS II Severovýchod                18827.8  
CZ.1.14 ROP Jihozápad                           18431.9  
CZ.1.15 ROP NUTS II Střední Čechy               17304.8  
CZ.1.25 OP Rybářství                              207.2  
CZ.2.16 OP Praha Konkurenceschopnost             6262.2  
CZ.2.17 OP Praha Adaptibilita                    1292.0  
CZ.3.22 OP ČR - Polsko                            518.9

In [65]:
prdf.to_csv(os.path.join(dvs, 'programy.csv'))

In [66]:
(df.loc[:, [celk, euz, erar]].groupby(erar).sum()/10**9).round(0)

Celkové zdroje - Sml/Dod  \
veřejná instituce                             
False                                 324.0   
True                                  851.0   

                   Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)  
veřejná instituce                                                      
False                                                          157.0   
True                                                           632.0

Jde tedy o cca 800 mld na projekty za 1.2 bilionu, kofinance jsou tedy zhruba třetinová.

In [67]:
df.loc[:, [kofi,erar]].groupby(erar).describe().unstack()

poměr vlastních financí (0-1)                           \
                                          count      mean       std  min   
veřejná instituce                                                          
False                                   24593.0  0.401020  0.250348  0.0   
True                                    42726.0  0.166738  0.182790  0.0   

                                                           
                        25%       50%       75%       max  
veřejná instituce                                          
False              0.141056  0.500000  0.600012  1.358232  
True               0.005986  0.114204  0.259389  1.000000

**chyba v mejch počtech (max > 1)**

## Nejvíc

### Veřejné

In [69]:
top = df.groupby([zad,ico]).sum().loc[:, [euz]].sort_values(by=euz, ascending=False)/10**9
top.round(1).head(20)

Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)
Žadatel                                            IČ žadatele                                                    
Ředitelství silnic a dálnic ČR                     65993390                                                  96.1 
Správa železniční dopravní cesty, státní organi... 70994234                                                  79.5 
MINISTERSTVO PRÁCE A SOCIÁLNÍCH VĚCÍ               00551023                                                  20.2 
Dopravní podnik hl.m. Prahy ,akciová společnost    00005886                                                  11.9 
Moravskoslezský kraj                               70890692                                                  10.9 
Masarykova univerzita                              00216224                                                  10.8 
MINISTERSTVO VNITRA                                00007064                                                   7.8 
Úřad práce ČR - Krajská pobočka v Brně, kontakt... 72496991                                                   7.7 
České dráhy, a.s.                                  70994226                                                   7.6 
Univerzita Palackého v Olomouci                    61989592                                                   7.5 
Kraj Vysočina                                      70890749                                                   6.9 
Fyzikální ústav AV ČR, v. v. i.                    68378271                                                   6.8 
Ministerstvo školství, mládeže a tělovýchovy       00022985                                                   6.3 
Vysoké učení technické v Brně                      00216305                                                   5.9 
Středočeský kraj                                   70891095                                                   5.7 
VYSOKÁ ŠKOLA BÁŇSKÁ-TECHNICKÁ UNIVERZITA OSTRAVA   61989100                                                   5.7 
Ústecký kraj                                       70892156                                                   4.4 
Jihočeský kraj                                     70890650                                                   4.4 
PARDUBICKÝ KRAJ                                    70892822                                                   4.4 
Správa a údržba silnic Jihomoravského kraje, př... 70932581                                                   4.1

In [72]:
top.head(1000).to_csv(os.path.join(dvs, 'top-verejne.csv'))

### Privátní

In [74]:
top_priv = df.loc[~df.loc[:, erar], :]
top_priv = top_priv.groupby([zad,ico]).sum().loc[:, [euz]].sort_values(by=euz, ascending=False)/10**9
top_priv.head(20)

Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)
Žadatel                                            IČ žadatele                                                    
TŘINECKÉ ŽELEZÁRNY, a. s.                          18050646                                              2.300128 
ArcelorMittal Ostrava a.s.                         45193258                                              1.676587 
Dolní oblast VÍTKOVICE                             75125285                                              1.273931 
Veolia Energie ČR, a.s.                            45193410                                              0.956849 
VÚTS, a.s.                                         46709002                                              0.770943 
Svaz VKMO s.r.o.                                   25572245                                              0.703694 
SVAZ MĚST A OBCÍ ČESKÉ REPUBLIKY                   63113074                                              0.657223 
Unipetrol výzkumně vzdělávací centrum, a.s.        62243136                                              0.587261 
Elektrárny Opatovice, a.s.                         28800621                                              0.582806 
VÝZKUMNÝ A ŠLECHTITELSKÝ ÚSTAV OVOCNÁŘSKÝ HOLOV... 25271121                                              0.581367 
Techmania Science Center o.p.s.                    26396645                                              0.571900 
Lovochemie, a.s.                                   49100262                                              0.539012 
CESNET - ZÁJMOVÉ SDRUŽENÍ PRÁVNICKÝCH OSOB         63839172                                              0.534798 
Kanonie premonstrátů Teplá                         00479365                                              0.496984 
COMTES FHT a.s.                                    26316919                                              0.467348 
Multifunkční centrum zámek Lednice, zájmové sdr... 75122979                                              0.455870 
Synthesia, a.s.                                    60108916                                              0.454878 
Terezín - město změny, zájmové sdružení právnic... 72045230                                              0.443990 
PrimeCell Therapeutics a.s.                        27660320                                              0.442951 
CGMC, družstvo                                     28094468                                              0.433479

In [76]:
top_priv.head(1000).to_csv(os.path.join(dvs, 'top-soukrome.csv'))

## Golf a hotely

In [77]:
len(df.loc[df.loc[:, nazev].str.contains('golf')])

6

In [78]:
df.loc[df.loc[:, 'Název projektu'].str.contains('golf')].sum()/10**6

Celkové zdroje - Sml/Dod                                                    254.482285
Rozhodnutí/Smlouva o poskytnutí dotace (celkové zdroje)                      99.682815
Rozhodnutí/Smlouva o poskytnutí dotace (veřejné prostředky celkem)           85.169353
Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)                           83.445630
Proplacené prostředky příjemcům - vyúčtované (veřejné prostředky celkem)     70.928785
Proplacené prostředky příjemcům - vyúčtované (EU zdroje)                     83.445630
Certifikované prostředky (veřejné prostředky celkem)                         70.928785
Pořadí v rámci v projektu (filtr)                                             0.000006
Počet míst realizace                                                          0.000008
veřejná instituce                                                             0.000001
poměr vlastních financí (0-1)                                                 0.000003
dtype: float64

## Neziskovky

**Nekompletní, bude až z právní formy z RES, tohle obšancovávání je nepřesný**

In [108]:
df.loc[df.loc[:, prfos] == 'Komory, profesní a zájmové sdružení', prfo].unique()

array(['Zájmové sdružení právnických osob', 'Spolek', 'Pobočný spolek',
       'Zájm.sdružení práv.osob', 'Komora s vyj.prof.komor',
       'Sdruž.(svaz,spolek,spol', 'Org.jednotka sdružení',
       'Stav.org.-profesní kom.', 'Komora (hospodářská, agrární)',
       'Mezinár. org.a sdružení', 'Zájmové sdružení',
       'Organizační jednotka občanského sdružení',
       'Mezinárodní nevládní organizace'], dtype=object)

In [104]:
nzs = df.loc[(df.loc[:, prfos] == 'Neziskový sektor') | (df.loc[:, prfo] == 'Ústav')]
nzs.sum()/10**6

Celkové zdroje - Sml/Dod                                                    12389.733936
Rozhodnutí/Smlouva o poskytnutí dotace (celkové zdroje)                     11048.307980
Rozhodnutí/Smlouva o poskytnutí dotace (veřejné prostředky celkem)           9527.357691
Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)                          10003.198079
Proplacené prostředky příjemcům - vyúčtované (veřejné prostředky celkem)     8625.621203
Proplacené prostředky příjemcům - vyúčtované (EU zdroje)                     6708.710529
Certifikované prostředky (veřejné prostředky celkem)                         5777.701690
Pořadí v rámci v projektu (filtr)                                               0.001609
Počet míst realizace                                                            0.005024
veřejná instituce                                                               0.000032
poměr vlastních financí (0-1)                                                   0.000181
dtype: float64

1600 projektů

In [105]:
nzs.shape

(1609, 30)

In [81]:
nejnzs = nzs.groupby(zad).sum().sort_values(ascending=False, by=euz)/10**6

nejnzs.loc[:, [euz]].head(10).round(0)

Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)
Žadatel                                                                                              
Techmania Science Center o.p.s.                                                                572.0 
Kanonie premonstrátů Teplá                                                                     497.0 
Římskokatolická farnost Velehrad                                                               408.0 
IQLANDIA, o.p.s.                                                                               407.0 
Federace židovských obcí v České republice                                                     285.0 
Energeia, o.p.s.                                                                               254.0 
Člověk v tísni, o.p.s.                                                                         244.0 
Benediktinské opatství sv. Václava v Broumově                                                  196.0 
Muzeum a společenské centrum Dobrovice, o.p.s.                                                 163.0 
Centrála cestovního ruchu Východní Moravy, o.p.s.                                              136.0

In [89]:
nejnzs.loc[:, [euz]].head(1000).to_csv(os.path.join(dvs, 'neziskovky.csv'))

neziskovky bez církví

In [91]:
nzsb = nzs.loc[~nzs.loc[:, prfo].str.contains('írk')]

In [92]:
dfn = pd.DataFrame(nzsb.groupby([prfo, zad]).sum().loc[:, euz]/10**6)
dfn.sort_values(by=euz, ascending=False).head(10).round(0)

Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)
Hosp.právní forma           Žadatel                                                                                               
Obecně prospěšná společnost Techmania Science Center o.p.s.                                                                 572.0 
                            IQLANDIA, o.p.s.                                                                                407.0 
                            Energeia, o.p.s.                                                                                254.0 
                            Člověk v tísni, o.p.s.                                                                          244.0 
                            Muzeum a společenské centrum Dobrovice, o.p.s.                                                  163.0 
                            Centrála cestovního ruchu Východní Moravy, o.p.s.                                               136.0 
                            OSTROV ZDRAVÍ o.p.s.                                                                            128.0 
Nadace                      ,, Nadace Josefa Viewegha Sklářská huť František "                                              127.0 
                            Nadace Partnerství (v mezinárodním styku "Czech...                                               92.0 
Obecně prospěšná společnost Vyšší odborná škola DAKOL a Střední škola DAKOL...                                               85.0

In [94]:
dfn.sort_values(by=euz, ascending=False).head(1000).to_csv(os.path.join(dvs, 'neziskovky-bez-cirkvi.csv'))

## Obce
per capita? jsou někde deviace?

Škromachovic

In [85]:
df.loc[:, zad].str.contains('Rohatec').sum()

4

In [86]:
df.loc[df.loc[:, nuts] == 'Rohatec'].sum()/10**6

Celkové zdroje - Sml/Dod                                                    326.411804
Rozhodnutí/Smlouva o poskytnutí dotace (celkové zdroje)                     183.536061
Rozhodnutí/Smlouva o poskytnutí dotace (veřejné prostředky celkem)          163.902748
Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)                          168.626124
Proplacené prostředky příjemcům - vyúčtované (veřejné prostředky celkem)    150.920879
Proplacené prostředky příjemcům - vyúčtované (EU zdroje)                    144.642873
Certifikované prostředky (veřejné prostředky celkem)                        129.313003
Pořadí v rámci v projektu (filtr)                                             0.000037
Počet míst realizace                                                          0.000037
veřejná instituce                                                             0.000011
poměr vlastních financí (0-1)                                                 0.000016
dtype: float64

In [87]:
(df.loc[df.loc[:, nuts] == 'Rohatec', [prog, euz, erar]].groupby([erar, prog]).sum()/10**6).unstack().round(0)

Rozhodnutí/Smlouva o poskytnutí dotace (EU zdroje)  \
Číslo a název programu                       CZ.1.02 OP Životní prostředí   
veřejná instituce                                                           
False                                                                15.0   
True                                                                 26.0   

                                                       \
Číslo a název programu CZ.1.03 OP Podnikání a inovace   
veřejná instituce                                       
False                                           111.0   
True                                              NaN   

                                                                \
Číslo a název programu CZ.1.04 OP Lidské zdroje a zaměstnanost   
veřejná instituce                                                
False                                                      4.0   
True                                                       NaN   

                                                             \
Číslo a název programu CZ.1.06 Integrovaný operační program   
veřejná instituce                                             
False                                                   NaN   
True                                                    0.0   

                                                                      \
Číslo a název programu CZ.1.07 OP Vzdělávání pro konkurenceschopnost   
veřejná instituce                                                      
False                                                            NaN   
True                                                             3.0   

                                                       
Číslo a název programu CZ.1.11 ROP NUTS II Jihovýchod  
veřejná instituce                                      
False                                             NaN  
True                                              9.0

In [88]:
ver.loc[:, 'NÁZEV'].str.contains('Holovo').sum()

0

### TODO
- <strike>kontroluj erár (po implementaci Ostatní) vůči MFČR seznamu (v dubnu novej)</strike>
- <strike>**dopravní podnik prahy je vedenej jako podnikatelskej subjekt**</strike>
 - <strike>45274649 28196678 26827719 00005886 25508881 61974757 - tohle jsou příklady IČA eráru, který jsou podnikatelé v datasetu</strike>
 - <strike>jinej mfcr seznam http://www.mfcr.cz/cs/verejny-sektor/rozpoctove-ramce-statisticke-informace/verejny-sektor/verejne-spolecnosti/2016/seznam-verejnych-spolecnosti-v-cr-2016-24752 -nezapomeň lpadnout nulama (třeba DPP je bez nul)</strike>
- vyřešit pořadí projektů - jsou ty skrytý řádky důležitý?
- <strike>roztřídit Ostatní mezi soukromý a veřejný (je tam toho pár)</strike>
- rozkouskovat na stages - příprava, realizace, zaplacení, certifikace
- všechny tabulky dělat pro všechny peněžní metriky
- <strike>Česko-Polsko vyřešit (je to ale jen cca 1 mld)</strike>
- <strike>uložit názvy sloupců do proměnných</strike>
- kofinance + histogram kofinančních poměrů
- <strike>přidat dummy sloupec - veřejné či ne</strike>
- <strike>rozdělení na operační programy</strike>
- dotace per capita (dotace pro veřejnost i soukromníky)
- Agrofert, jako minule, ze závěrky (+ ARES); + pivovary, škoda, soukromý sš/vš, učiliště, ...; scio; křetinský, penta, ppf atd., soláry
- doprava - SŽDC, ŘSD atd.
- CzechTrade, Czech Tourism atd.; Odbory
- <strike>rozlišit v rámci neziskovek na církve, ops, ...</strike>
- uklidit
 - <strike>např. obecně prospěšný společnosti (viz výše)</strike>
 - <strike>VŠ a VŠ (státní) jsou zvlášť, ale ve skutečnosti jsou jedno (soukromý VŠ jsou v podnikatelích)</strike>
 - <strike>Ostatní -> ústav zařadit mezi neziskovky (je to forma ops)</strike>
- <strike>najít firmy s významným podílem státu (ČD, Ruzyň, ČEZ)</strike>
- <strike>IČO a názvů žadatelů jsou různý počty - unifikuj</strike>
- podnikatelský subjekty rozkouskovat podle Čísla prioritní osy (a/nebo Čísla oblasti podpory) - 3.2.2 je CZC nebo Zoner

- <strike>len(ico) != 8 vyhodit, spocitat jejich miliardy</strike>
- najit mezi komorama neziskovky
- najit neziskovky a zahranicne vlastneny podniky v ESA klasifikaci v RES
- group by pravni forma (ESA z RES) a NACE (RES)
